### ISSUE TITLE
On next branch, using Helm for completion often fails

### ISSUE THREAD

**Issue by jwiegley** [magit/magit#1770]
I have `(setq magit-completing-read-function (quote helm-completing-read-with-cands-in-buffer))` in my `.emacs`, but this very often results in:

```
magit-completing-read: Wrong number of arguments: (10 . 10), 7
```

For all kinds of operations.  At the moment, it's happening whenever I try to do `magit-ediff-stage`.

If it doesn't happen for you, let me know and I'll track it down further.

**Comment by tarsius** [comment]
Magit  used to provide some `completing-read` variant for helm. But it was broken and I also could not figure out quickly how to instead use some function that comes with helm itself. I don't remember the details.

I used helm when it first came out, while lately I have used ido. I plan to give helm another change, but most likely after I have released 2.1.0.

I had a quick look now though, and thing you should probably use `helm--completing-read-default` instead. Otherwise please post a backtrace.

**Comment by jwiegley** [comment]
Yes, that worked quite nicely, thank you.
